src/System/Wlog/Severity.hs

```diff
@@ -26,5 +39,56 @@ data Severity
 instance FromJSON Severity
 instance ToJSON   Severity
 
+-- | Set of 'Severity'.
+type Severities = Set Severity
+
 -- | Internal type of log records.
 data LogRecord = LR !Severity !Text deriving Show
+
+-- | 'Set' of all 'Severity's.
+allSeverities :: Set Severity
+allSeverities = Set.fromList [Debug, Info, Notice, Warning, Error]
+
+-- | Returns the 'Set' of 'Severity's of elements greater than the given value.
+severityPlus :: Severity -> Set Severity
+severityPlus s = Set.filter (>= s) allSeverities
```

Comment:
Another possible implementation of this function is to use `Enum` and `Bounded` instances for `Severity`:

```haskell
severityPlus s = Set.fromAscList [s .. maxBound]
```